core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -355,4 +329,38 @@ private static String featureName(final URL url) {
         // fallback
         return Math.abs(url.hashCode()) + Features.EXTENSION;
     }
+
+    private static boolean isCukeSpaceTest(final TestClass testClass, final ClassLoader loader) {
+        Class<? extends Annotation> runWithType = null;
+        try {
+            runWithType = (Class<? extends Annotation>) loader.loadClass("org.junit.runner.RunWith");
+
+        } catch (final ClassNotFoundException error) {
+            // no-op
+        } catch (final NoClassDefFoundError e) {
+            // no-op
+        }
+
+        if (runWithType == null) {
+            return false;
+        }
+        final Annotation runWith = testClass.getAnnotation(runWithType);
+        if (runWith == null) {
+            return false;
+        }
+
+        try {
+            final Class<?> runner = Class.class.cast(runWithType.getMethod("value").invoke(runWith));
+            if ((!ArquillianCucumber.class.equals(runner) && !CukeSpace.class.equals(runner))) {

```

Comment:
skipEnrichment (trying to make it less technical or linked to code to ensure it can be maintained even with a huge refactoring)?

For helper i would say we can probably refactor the processor but would need to know the details exactly to say anytging more relevant. Finders are already static normally but we can create some kind of services.